src/main/java/cart/domain/Order.java

```diff
@@ -0,0 +1,63 @@
+package cart.domain;
+
+import java.util.List;
+
+public class Order {
+    private final Member member;
+    private final List<OrderItem> orderItems;
+    private final MemberCoupon memberCoupon;
+    private Long id;
+
+    public Order(Member member, List<OrderItem> orderItems, MemberCoupon memberCoupon) {
+        this.member = member;
+        this.orderItems = orderItems;
+        this.memberCoupon = memberCoupon;
+    }
+
+    public Order(Long id, Member member, List<OrderItem> orderItems, MemberCoupon memberCoupon) {
+        this.id = id;
+        this.member = member;
+        this.orderItems = orderItems;
+        this.memberCoupon = memberCoupon;
+    }
+
+    public Money getTotalPrice() {
+        return new Money(orderItems.stream().mapToInt(OrderItem::getPrice).sum());
+    }
+
+    public Money getDiscountPrice() {
+        Money totalPrice = getTotalPrice();
+        return Money.subtract(totalPrice, getDiscountedPrice());
+    }
+
+    public Money getShippingFee() {
+        return ShippingFeePolicy.findShippingFee(getTotalPrice());
+    }
+
+    public Money getDiscountedPrice() {
+        if (memberCoupon == null) {
+            return new Money(0);
+        }
```

Comment:
```suggestion
        if (memberCoupon == null) {
            return Money.noDiscount();
        }
```

요런 식으로 정적 팩토리 메서드를 활용할 수도 있어요.

Reply:
네! 감사합니다!!

Reply:
오 Money.zero() 좋네요 👍